app/controllers/schools_controller.rb

```diff
@@ -104,9 +106,127 @@ def update_code_of_conduct
     redirect_to standing_school_path
   end
 
+  # GET /school/discord
+  def discord_configuration
+    authorize current_school
+  end
+
+  # GET /school/discord_server_roles?cause=
+  def discord_server_roles
+    authorize current_school
+
+    if params[:cause].present? && params[:cause] == "DeletedRoles"
+      @sync_service = Discord::SyncRolesService.new(school: current_school)
+    end
+  end
+
+  # PATCH /school/discord_configuration
+  def discord_credentials
+    authorize current_school
+
+    bot_token = params["bot_token"]
+
+    discord_config = current_school.configuration.dig("discord") || {}
+    discord_config["server_id"] = params["server_id"]
+    discord_config["bot_user_id"] = params["bot_user_id"]
+    discord_config["bot_token"] = bot_token.presence ||
+      discord_config.dig("bot_token")
+
+    current_school.update!(
+      configuration:
+        current_school.configuration.merge({ "discord" => discord_config })
+    )
+
+    flash[:success] = I18n.t(
+      "schools.discord_credentials.discord_config_stored"
+    )
+
+    redirect_to discord_configuration_school_path
+  end
+
+  # POST /school/discord_sync_roles
+  def discord_sync_roles
+    authorize current_school
+
+    role_sync_service = Discord::SyncRolesService.new(school: current_school)
+
+    if role_sync_service.sync_ready? &&
+         (!role_sync_service.deleted_roles? || params[:confirmed_sync].present?)
+      role_sync_service.sync
+      flash[:success] = I18n.t(
+        "schools.discord_sync_roles.sync_service_result.success"
+      )
+    elsif role_sync_service.deleted_roles?
+      redirect_to discord_server_roles_school_path(cause: "DeletedRoles")
+      flash[:warn] = I18n.t(
+        "schools.discord_sync_roles.sync_service_result.warn"
+      )
+
+      return
+    else
+      flash[:error] = I18n.t(
+        "schools.discord_sync_roles.sync_service_result.warn",
+        error_msg: role_sync_service.error_message
+      )
+    end
+
+    redirect_to discord_server_roles_school_path
+  end
+
+  # POST /school/update_default_discord_roles
+  def update_default_discord_roles
+    authorize current_school
+
+    discord_role_ids =
+      current_school
+        .discord_roles
+        .where(id: params[:default_role_ids])
+        .pluck(:discord_id)
+
+    config =
+      current_school.configuration.deep_merge(
+        { "discord" => { "default_role_ids" => discord_role_ids } }
+      )
```

Comment:
It seems quite odd to store the `default` property of the stored discord role inside the current school configuration (the old method). Shouldn't this be moved into a `default` boolean column inside the `discord_roles` table?